backend/src/agents/generalist_agent.py

```diff
@@ -0,0 +1,39 @@
+import logging
+from src.prompts import PromptEngine
+from src.agents import ChatAgent, chat_agent
+from src.utils import Config
+from src.utils.web_utils import (
+    answer_user_question,
+)
+import json
+
+logger = logging.getLogger(__name__)
+config = Config()
+
+engine = PromptEngine()
+
+
+@chat_agent(
+    name="GeneralistAgent",
+    description="This agent attempts to answer a general question using only the llm",
+    tools=[],
+)
+class GeneralistAgent(ChatAgent):
+    async def invoke(self, utterance) -> str:
+        try:
+            answer_to_user = await answer_user_question(utterance, self.llm, self.model)
+            answer_result = json.loads(answer_to_user)
+            if answer_result["status"] == "error":
```

Comment:
I don't see the value in the error handling of `answer_user_question`, again it looks to be error handling the llm call itself. I would get rid of this if you want